sklearn/__init__.py

```diff
@@ -16,10 +16,12 @@
 import re
 import warnings
 
+from .exceptions import SklearnDeprecationWarning
 
 # Make sure that DeprecationWarning within this package always gets printed
 warnings.filterwarnings('always', category=DeprecationWarning,
```

Comment:
Hm... I think if we use a custom warning, we should use it everywhere. (Its a single line to search and replace them all).
Not sure if @jnothman and @GaelVaroquaux agree though.